includes/Classifai/Features/AudioTranscriptsGeneration.php

```diff
@@ -348,6 +348,189 @@ public function add_transcription( string $text = '', int $attachment_id = 0 ) {
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+
+		if ( empty( $body ) ) {
+			return new WP_Error( 'empty_download', __( 'Downloaded file is empty.', 'classifai' ) );
+		}
+
+		$filename = wp_basename( wp_parse_url( $url, PHP_URL_PATH ) );
+
+		if ( empty( $filename ) ) {
+			$filename = 'audio_' . time() . '.tmp';
+		}
+
+		$temp_file_path = $temp_dir . $filename;
+
+		global $wp_filesystem;
+
+		// Initialize the WordPress filesystem.
+		if ( ! $wp_filesystem ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+			WP_Filesystem();
+		}
+
+		$bytes_written = $wp_filesystem->put_contents(
+			$temp_file_path,
+			$body
+		);
+
+		if ( false === $bytes_written ) {
+			return new WP_Error( 'file_write_failed', __( 'Failed to write temporary file.', 'classifai' ) );
+		}
+
+		// Step 5: Validate audio file type
+		$supported_audio_extensions = [
+			'mp3',
+			'mp4',
+			'mpeg',
+			'mpga',
+			'm4a',
+			'wav',
+			'webm',
+		];
+
+		$supported_audio_mime_types = [
+			'audio/mpeg',
+			'audio/mp4',
+			'audio/x-m4a',
+			'audio/wav',
+			'audio/x-wav',
+			'audio/x-pn-wav',
+			'audio/webm',
+		];
+
+		$filetype  = wp_check_filetype( $temp_file_path );
+		$extension = strtolower( $filetype['ext'] ?? '' );
+		$mime_type = $filetype['type'] ?? '';
+
+		if ( empty( $extension ) || empty( $mime_type ) ) {
+			return new WP_Error( 'filetype_unknown', __( 'Could not determine file type.', 'classifai' ) );
+		}
+
+		if ( ! in_array( $extension, $supported_audio_extensions, true ) || ! in_array( $mime_type, $supported_audio_mime_types, true ) ) {
+			return new WP_Error(
+				'unsupported_audio_type',
+				sprintf(
+					/* translators: %1$s Extension, %2$s MIME Type. */
+					__( 'Unsupported audio type. Extension: %1$s, MIME type: %2$s', 'classifai' ),
+					$extension,
+					$mime_type
+				)
+			);
+		}
+
+		// Just an additional check to keep things secure.
+		if ( function_exists( 'finfo_open' ) ) {
+			$finfo          = finfo_open( FILEINFO_MIME_TYPE );
+			$real_mime_type = finfo_file( $finfo, $temp_file_path );
+			finfo_close( $finfo );
+
+			if ( ! in_array( $real_mime_type, $supported_audio_mime_types, true ) ) {
+				return new WP_Error(
+					'unsupported_audio_content',
+					sprintf(
+						/* translators: %1$s Detected MIME Type. */
+						__( 'File content does not match supported audio types. Detected: %s', 'classifai' ),
+						$real_mime_type
+					)
+				);
+			}
+		}
+
+		return $temp_file_path;
+	}
+
+	/**
+	 * Returns if the resource is of type attachment, URL or system path.
+	 *
+	 * @param int|string $audio_resource The ID of the attachment, or system or URL path
+	 *                                   to the audio file.
+	 *
+	 * @return boolean|string The resource type. (attachment, url, path or false if none).
+	 */
+	public static function get_resource_type( $audio_resource ) {
```

Comment:
I think all these methods (this and below) could be moved to our `Helpers.php` file, as they may be used in the future in other places (they all seem generic enough to not need to live here)